File: baekJoon/baekjoon_20164.cpp
// 홀수 홀릭 호석

#include <iostream>
#include <string>
#include <limits.h>
#include <algorithm>

using namespace std;


int max_v = INT_MIN, min_v = INT_MAX;
string n;

void dfs(string str, int s) {
	int cnt = 0;
	for (auto ch : str)
		if ((ch - '0') % 2 == 1) cnt++;
	if (str.size() == 1) {
		min_v = min(min_v, cnt + s);
		max_v = max(max_v, cnt + s);
		return;
	}
	if (str.size() == 2) {
		int next_num = (str[0] - '0') + (str[1] - '0');
		dfs(to_string(next_num), cnt + s);
	}
	else {
		for (int i = 1; i < str.size() - 1; i++) {
			for (int j = i + 1; j < str.size(); j++) {
				string next1 = str.substr(0, i);
				string next2 = str.substr(i, j - i);
				string next3 = str.substr(j);
				int next_num = stoi(next1) + stoi(next2) + stoi(next3);
				dfs(to_string(next_num), cnt + s);
			}
		}
	}
}

int main() {
	cin.sync_with_stdio(0);
	cin.tie(0);
	cin >> n;
	dfs(n, 0);
	cout << min_v << " " << max_v << '\n';
	return 0;
}

File: baekJoon/baekjoon_20164.py
# 홀수 홀릭 호석

import sys
input = sys.stdin.readline
N = str(input().rstrip('\n'))
max_v = float('-inf')
min_v = float('inf')

def dfs(n, s):
    cnt = 0
    for idx in n:
        if int(idx) % 2 == 1:
            cnt += 1
    if len(n) == 1:
        global max_v, min_v
        max_v = max(max_v, s + cnt)
        min_v = min(min_v, s + cnt)
        return
    if len(n) == 2:
        next = int(n[0]) + int(n[1])
        dfs(str(next), s + cnt)
        return
    else:
        for i in range(1, len(n) - 1):
            for j in range(i + 1, len(n)):
                next1 = int(n[:i])
                next2 = int(n[i:j])
                next3 = int(n[j:])
                dfs(str(next1 + next2 + next3), s + cnt)

dfs(N, 0)
print(min_v, max_v)